org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/refactoring/GetRefactorEditHandlerTest.java

```diff
@@ -88,6 +93,47 @@ public void testExtractVariable() throws Exception {
 		Assert.assertEquals(1, refactorEdit.command.getArguments().size());
 	}
 
+	@Test
+	public void testExtractVariableFinal() throws Exception {
+		Preferences mockPreferences = mock(Preferences.class);
+		when(JavaLanguageServerPlugin.getPreferencesManager().getPreferences()).thenReturn(mockPreferences);
+		when(mockPreferences.getCodeGenerationAddFinalForNewDeclaration()).thenReturn("all");

```

Comment:
I think you can actually just set `preferences.setCodeGenerationAddFinalForNewDeclaration("all");` here (same for the testcase below).

This is because there is a `preferences` field defined in `AbstractProjectsManagerBasedTest`, which is in the hierarchy of this class (AbstractSelectionTest -> AbstractSelectionTest -> AbstractProjectsManagerBasedTest). If you look at `initPreferenceManager`, and some of the other classes there, you'll see they set up pretty much exactly what you're doing here.